tests/model/SiteTreeTest.php

```diff
@@ -1334,25 +1333,73 @@ public function testCanPublish() {
 		$this->assertFalse($page->canPublish());
 	}
 
+	/**
+	 * Test that the controller name for a SiteTree instance can be gathered by appending "Controller" to the SiteTree
+	 * class name in a PSR-2 compliant manner.
+	 */
+	public function testGetControllerName()
+	{
+		$class = new Page;
+		$this->assertSame('PageController', $class->getControllerName());
+	}
+
+	/**
+	 * Test that underscored class names (legacy) are still supported, but raise a deprecation notice. The exception and
+	 * manual catching is in place to ensure that we can (A) detect that a deprecation notice was raised and (B) that
+	 * we can restore the old error handling after catching the exception we threw from this test. Simply asserting that
+	 * an exception is thrown will not allow further logic to execute after the exception is thrown.
+	 */
+	public function testGetControllerNameWithUnderscoresIsSupported()
+	{
+		$defaultEnabled = Deprecation::get_enabled();
+		Deprecation::set_enabled(true);
+		Deprecation::notification_version('4.0.0');
+
+		// Catch user_errors thrown by deprecation
+		set_error_handler(
```

Comment:
I don't think it does automatically, needs an arg set in the config to raise errors etc as exceptions. Adding now.